src/sentry/static/sentry/app/components/forms/numberField.jsx

```diff
@@ -8,8 +8,18 @@ export default class NumberField extends InputField {
     max: PropTypes.number
   };
 
-  coerceValue(value) {
-    return parseInt(value, 10);
+  coerceValue(value, prevValue) {
+    let intValue = parseInt(value, 10);
+
+    // return previous value if new value is NaN, otherwise, will get recursive error
+    // eslint-disable-next-line no-self-compare
+    let isNewCoercedNaN = intValue !== intValue;
```

Comment:
did you consider just using isNaN here? i guess this could theoretically catch other problems?

Reply:
https://stackoverflow.com/a/32884332/3842656

Reply:
`isNaN` should work since we're checking output of `parseInt`